atd-etl/cris_import/cris_import.py

```diff
@@ -0,0 +1,821 @@
+#!/usr/bin/python3
+
+import os
+import re
+import time
+import hashlib
+import datetime
+import tempfile
+from subprocess import Popen, PIPE
+
+import shutil
+import boto3
+import sysrsync
+import psycopg2
+import psycopg2.extras
+import onepasswordconnectsdk
+from sshtunnel import SSHTunnelForwarder
+from onepasswordconnectsdk.client import Client, new_client
+
+import lib.mappings as mappings
+import lib.sql as util
+import lib.graphql as graphql
+from lib.helpers_import import insert_crash_change_template as insert_change_template
+
+DEPLOYMENT_ENVIRONMENT = os.environ.get(
+    "ENVIRONMENT", "development"
+)  # our current environment from ['production', 'development']
+ONEPASSWORD_CONNECT_TOKEN = os.getenv("OP_API_TOKEN")  # our secret to get secrets 🤐
+ONEPASSWORD_CONNECT_HOST = os.getenv("OP_CONNECT")  # where we get our secrets
+VAULT_ID = os.getenv("OP_VAULT_ID")
+
+
+def main():
+    secrets = get_secrets()
+
+    # 😢 why not `global variable = value`??
```

Comment:
@mddilley wrote:

> This global keyword also caught my eye 👀 Is it possible to avoid using globals by declaring secrets at the top level and then assigning it here on line 34? Not sure if there is a limitation with that route.

I know, it's crazy you can't `global variable_name = value` to me. But i've found that either you `global` them while not in the global scope and then define them, or you define them in the global scope as you suggested. I'm trying to get in the habit of doing as little code as possible in my python outside of functions because I feel like it's good style - such that one's python programs can be python libraries without modification just by importing them. 

I waffled back and forth on this real hard, but I decided I wanted to go `global` definitions in the `main()` function so that they are entirely scoped & defined in single section of the file. I realize that because `main()` sits at the top of the file, this isn't too far away from where you'd scope them in practice, but "outside vs. inside a function" seems like a significant separation of context. Imagine that `main()` were defined as the last function of the file, which is equally valid, then maybe this choice makes sense? I know I'm like beyond splitting hairs, so please don't take me too seriously here!

Either way - it's not pretty, and I'd be so happy to conform to the pattern you suggest if you think that's better after reading this. What feels best to you?